File: etcp-customer-interface/src/main/java/com/etcp/customer/service/CustomerService.java
package com.etcp.customer.service;

import com.etcp.pojo.TbCustomer;
import entity.Password;
import entity.PageResult;

import java.util.List;
/**
 * 业务逻辑接口
 * @author Steven
 *
 */
public interface CustomerService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbCustomer> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbCustomer customer);
	
	
	/**
	 * 修改
	 */
	public void update(TbCustomer customer);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbCustomer findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbCustomer customer, int pageNum, int pageSize);
	/**
	 * 更新状态
	 * @param customerId
	 * @param status   状态(0:未审核，1：审核通过,2:审核不通过，3关闭店铺)
	 */
	public void updateStatus(Long customerId, Integer status);

	/**
	 * 修改密码
	 * @param password
	 * @return
	 */
	public boolean updatePassword(Password password);
	
}

File: etcp-admin-web/src/main/webapp/js/service/customerService.js
//服务层
app.service('customerService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../customer/findAll.do');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../customer/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../customer/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../customer/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../customer/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../customer/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../customer/search.do?page='+page+"&rows="+rows, searchEntity);
	}

    //状态(0:未审核，1：审核通过,2:审核不通过，3关闭店铺)
    this.updateState=function(customerId,status){
        return $http.get('../customer/updateStatus.do?customerId='+customerId + '&status='+status);
    }
});
